File: binding/src/libqwaitcondition.cpp
#include <QDeadlineTimer>
#include <QMutex>
#include <QReadWriteLock>
#include <QWaitCondition>
#include <qwaitcondition.h>
#include "libqwaitcondition.h"
#include "libqwaitcondition.hxx"

QWaitCondition* QWaitCondition_new() {
	 return new QWaitCondition();
}

bool QWaitCondition_Wait(QWaitCondition* self, QMutex* lockedMutex) {
	return self->wait(lockedMutex);
}

bool QWaitCondition_Wait2(QWaitCondition* self, QMutex* lockedMutex, unsigned long time) {
	return self->wait(lockedMutex, time);
}

bool QWaitCondition_Wait3(QWaitCondition* self, QReadWriteLock* lockedReadWriteLock) {
	return self->wait(lockedReadWriteLock);
}

bool QWaitCondition_Wait4(QWaitCondition* self, QReadWriteLock* lockedReadWriteLock, unsigned long time) {
	return self->wait(lockedReadWriteLock, time);
}

void QWaitCondition_WakeOne(QWaitCondition* self) {
	self->wakeOne();
}

void QWaitCondition_WakeAll(QWaitCondition* self) {
	self->wakeAll();
}

void QWaitCondition_NotifyOne(QWaitCondition* self) {
	self->notify_one();
}

void QWaitCondition_NotifyAll(QWaitCondition* self) {
	self->notify_all();
}

bool QWaitCondition_Wait22(QWaitCondition* self, QMutex* lockedMutex, QDeadlineTimer* deadline) {
	return self->wait(lockedMutex, *deadline);
}

bool QWaitCondition_Wait23(QWaitCondition* self, QReadWriteLock* lockedReadWriteLock, QDeadlineTimer* deadline) {
	return self->wait(lockedReadWriteLock, *deadline);
}

void QWaitCondition_Delete(QWaitCondition* self) {
    delete self;
}

File: binding/include/libqwaitcondition.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qwaitcondition.html
pub const qwaitcondition = struct {

    /// New constructs a new QWaitCondition object.
    pub fn New() QtC.QWaitCondition {
        return qtc.QWaitCondition_new();
    }


    pub fn Wait(self: ?*anyopaque, lockedMutex: ?*anyopaque) bool {
        return qtc.QWaitCondition_Wait(@ptrCast(self), @ptrCast(lockedMutex));
    }

    pub fn Wait2(self: ?*anyopaque, lockedMutex: ?*anyopaque, time: unsigned long) bool {
        return qtc.QWaitCondition_Wait2(@ptrCast(self), @ptrCast(lockedMutex), time);
    }

    pub fn Wait3(self: ?*anyopaque, lockedReadWriteLock: ?*anyopaque) bool {
        return qtc.QWaitCondition_Wait3(@ptrCast(self), @ptrCast(lockedReadWriteLock));
    }

    pub fn Wait4(self: ?*anyopaque, lockedReadWriteLock: ?*anyopaque, time: unsigned long) bool {
        return qtc.QWaitCondition_Wait4(@ptrCast(self), @ptrCast(lockedReadWriteLock), time);
    }

    pub fn WakeOne(self: ?*anyopaque, ) void {
        qtc.QWaitCondition_WakeOne(@ptrCast(self));
    }

    pub fn WakeAll(self: ?*anyopaque, ) void {
        qtc.QWaitCondition_WakeAll(@ptrCast(self));
    }

    pub fn NotifyOne(self: ?*anyopaque, ) void {
        qtc.QWaitCondition_NotifyOne(@ptrCast(self));
    }

    pub fn NotifyAll(self: ?*anyopaque, ) void {
        qtc.QWaitCondition_NotifyAll(@ptrCast(self));
    }

    pub fn Wait22(self: ?*anyopaque, lockedMutex: ?*anyopaque, deadline: QtC.QDeadlineTimer) bool {
        return qtc.QWaitCondition_Wait22(@ptrCast(self), @ptrCast(lockedMutex), @ptrCast(deadline));
    }

    pub fn Wait23(self: ?*anyopaque, lockedReadWriteLock: ?*anyopaque, deadline: QtC.QDeadlineTimer) bool {
        return qtc.QWaitCondition_Wait23(@ptrCast(self), @ptrCast(lockedReadWriteLock), @ptrCast(deadline));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QWaitCondition_Delete(@ptrCast(self));
    }
};
